File: sort/c_implement/test_array.c
//
// Created by kaku on 2019/10/6.
//

#include <stdlib.h>
#include "array.h"

#define ARRAY_SIZE 10

int main() {
    int *ptr;

    ptr = random_array(ARRAY_SIZE);
    show_array(ptr, ARRAY_SIZE, "Test Case.");
    free(ptr);

    return 0;
}

File: sort/py_impl/test_array.py
#!/usr/bin/env python3
# coding: utf-8
#
# author: kaku
# date: 19/10/07
#
# GitHub:
#
#   https://github.com/kakukosaku
#
# © 2019-2022 Kaku Kosaku All Rights Reserved
#
# Usage:
#   python -m py_impl.test_array

from .array import random_array, show_array, check_sorted

ARRAY_SIZE = 10


def test():
    arr = random_array(ARRAY_SIZE, (0, 100))
    show_array(arr, "Test Random Case")
    check_sorted(arr)
    arr = list(range(ARRAY_SIZE))
    show_array(arr, "Test Sorted Case")
    check_sorted(arr)


if __name__ == '__main__':
    test()
